sklearn/decomposition/_base.py

```diff
@@ -148,16 +160,15 @@ def inverse_transform(self, X):
         If whitening is enabled, inverse_transform will compute the
         exact inverse operation, which includes reversing whitening.
         """
+        xp, _ = get_namespace(X)
+
         if self.whiten:
-            return (
-                np.dot(
-                    X,
-                    np.sqrt(self.explained_variance_[:, np.newaxis]) * self.components_,
-                )
-                + self.mean_
+            scaled_components = (
+                xp.sqrt(self.explained_variance_[:, None]) * self.components_
```

Comment:
I think I observed a problem with `xp.newaxis` (`AttributeError`) for one of the libraries but I don't remember which. I will try again and put a comment if that's still the case.

Reply:
Oh! I just realized you wrote `np.newaxis` instead of `xp.newaxis`. Indeed you can keep `np.newaxis` since it is actually just a reference to `None` with a more explicit name, although it's a bit confusing to mix `xp` and `np` namespaces in the same snippet. But maybe this is fine here.

Reply:
I pushed a change to use `np.newaxis` whenever possible.